samples/Common/cpp/CameraHelper_cppwinrt.cpp

```diff
@@ -73,44 +74,41 @@ void CameraHelper::Initialize()
     m_mediaCapture.Failed({ this, &CameraHelper::MediaCapture_Failed });
 
+    // This call will throw if there are no cameras attached
     m_mediaCapture.InitializeAsync(mediaCaptureInitializationSettings).get();
 
-    // Get a list of available Frame source and iterate through them to find a video preview or 
+    // Get a list of available frame sources and iterate through them to find a video preview or 
     // a video record source with color images (and not IR, depth or other types)
     auto frameSourceIterator = m_mediaCapture.FrameSources().First();
     const wchar_t* cameraName;
-    do
+
+    // Look first for a MediaStreamType::VideoPreview source
+    auto mediaStreamTypeToLookFor = MediaStreamType::VideoPreview;
+    while (frameSourceIterator.HasCurrent())
     {
         auto mediaStreamType = frameSourceIterator.Current().Value().Info().MediaStreamType();
         auto sourceKind = frameSourceIterator.Current().Value().Info().SourceKind();
         cameraName = frameSourceIterator.Current().Value().Info().DeviceInformation().Name().c_str();
 
         std::wcout << cameraName << L" | MediaStreamType:" << (int)mediaStreamType << L" MediaFrameSourceKind:" << (int)sourceKind << std::endl;
-        if ((mediaStreamType == MediaStreamType::VideoPreview) && (sourceKind == MediaFrameSourceKind::Color))
+        if ((mediaStreamType == mediaStreamTypeToLookFor) && (sourceKind == MediaFrameSourceKind::Color))
         {
             break;
         }
-    } while (frameSourceIterator.MoveNext());
 
-    if (!frameSourceIterator.HasCurrent())
-    {
-        frameSourceIterator = m_mediaCapture.FrameSources().First();
-        do
+        // If we reach the end of the collection and did not find a source with MediaStreamType::VideoPreview
+        if (!frameSourceIterator.MoveNext() && mediaStreamTypeToLookFor == MediaStreamType::VideoPreview)
         {
-            auto mediaStreamType = frameSourceIterator.Current().Value().Info().MediaStreamType();
-            auto sourceKind = frameSourceIterator.Current().Value().Info().SourceKind();
-            cameraName = frameSourceIterator.Current().Value().Info().DeviceInformation().Name().c_str();
-
-            if ((mediaStreamType == MediaStreamType::VideoRecord) && (sourceKind == MediaFrameSourceKind::Color))
-            {
-                break;
-            }
-        } while (frameSourceIterator.MoveNext());
+            // Look for a MediaStreamType::VideoRecord source (1-pin cameras often expose only this type)
+            mediaStreamTypeToLookFor = MediaStreamType::VideoRecord;
+            frameSourceIterator = m_mediaCapture.FrameSources().First();
+        }
     }
+
     if (!frameSourceIterator.HasCurrent())
     {
         std::cerr << "No valid video frame sources were found with source type color.";
-        throw new winrt::hresult(MF_E_INVALIDMEDIATYPE);
+        winrt::throw_hresult(winrt::hresult(MF_E_INVALIDMEDIATYPE));
```

Comment:
nit: should be able to simplify to `winrt::throw_hresult(MF_E_INVALIDMEDIATYPE);`